Multisig/Logic/Models/KeyInfo.swift

```diff
@@ -42,12 +42,14 @@ extension KeyInfo {
     }
 
     /// Returns number of existing key infos
-    static var count: Int {
+    static func count(_ type: KeyType? = nil) -> Int {
         do {
-            let context = App.shared.coreDataStack.viewContext
-            let fr = KeyInfo.fetchRequest().all()
-            let itemCount = try context.count(for: fr)
-            return itemCount
+            let items = try KeyInfo.all()
+            if let type = type {
+                return items.filter({ keyInfo in keyInfo.keyType == type }).count
```

Comment:
Just an idea, I think it makes sense to execute a "count" fetch request which will return the count from the database, no need to count manually.

But this would be more complicated. Until we see this has performance issues, I'm ok with this implementation.